### PR TITLE
Session.virtualfile_to_dataset: Add new parameters 'dtype'/'index_col' for pandas output

### PR DESCRIPTION
**Pull request by seisman**
Address https://github.com/GenericMappingTools/pygmt/pull/3131#discussion_r1536934045.

Changes in this PR:

- `GMT_DATASET.to_dataframe`: Add new parameters `names`/`dtype`/`index_col`
- `Session.virtualfile_to_dataset`: AAdd new parameter `dtype`/`index_col` and rename `column_names` to `names`
- Simplify some wrappers

The parameter names `names`/`dtype`/`index_col` are very close to the ones in the [`pd.read_csv`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html) function.

### CODE REVIEW

File: pygmt/clib/session.py [hunk 1]
```diff
@@ -1747,6 +1747,8 @@ def virtualfile_to_dataset(
         vfname: str,
         output_type: Literal["pandas", "numpy", "file"] = "pandas",
         column_names: list[str] | None = None,
```
Comment (seisman) [inline pygmt/clib/session.py:1749]: In `Session.virtualfile_to_dataset`, do we want to rename `column_names` to `names`?

Comment (seisman) [reply pygmt/clib/session.py]: Done in f211b7d69c4701adf11bf2a0fb4226aa1265c35b.

File: pygmt/datatypes/dataset.py [hunk 2]
```diff
@@ -211,5 +226,11 @@ def to_dataframe(self) -> pd.DataFrame:
                 pd.Series(data=np.char.decode(textvector), dtype=pd.StringDtype())
             )
 
-        df = pd.concat(objs=vectors, axis=1)
+        df = pd.concat(objs=vectors, axis="columns")
+        if names is not None:  # Assigne column names
```
Comment (yvonnefroehlich) [inline pygmt/datatypes/dataset.py:230]:
```suggestion
        if names is not None:  # Assign column names
```

File: pygmt/clib/session.py (hunk 1 above)
Comment (seisman) [reply pygmt/clib/session.py]: Do you mean using `column_names` in both `GMT_DATASET.to_dataframe` and `Session.virtualfile_to_dataset`?

Comment (weiji14) [reply pygmt/clib/session.py]: Yes, i.e. revert https://github.com/GenericMappingTools/pygmt/commit/f211b7d69c4701adf11bf2a0fb4226aa1265c35b

**Review by weiji14 — APPROVED** [review]

**Pull request merged**